servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -6844,4 +6844,24 @@ boolean isAllowedPutMembership(Principal principal, final AthenzDomain domain, f
         return false;
     }
 
+    @Override
+    public DomainRoleMembership getPendingDomainRoleMembersList(ResourceContext ctx) {
+        final String caller = "getpendingdomainrolememberslist";
+        metric.increment(ZMSConsts.HTTP_GET);
+        metric.increment(ZMSConsts.HTTP_REQUEST);
+        metric.increment(caller);
+        final String principalDomain = getPrincipalDomain(ctx);
+        Object timerMetric = metric.startTiming("getpendingdomainrolememberslist_timing", null, principalDomain);
+        logPrincipal(ctx);
+        validateRequest(ctx.request(), caller);
+        Principal ctxPrincipal = ((RsrcCtxWrapper) ctx).principal();
```

Comment:
since we get the principal object only for debug we should included it within the if block or just reference it directly in the debug call